xbmc/Application.cpp

```diff
@@ -2778,6 +2779,26 @@ bool CApplication::OnMessage(CGUIMessage& message)
     }
     break;
 
+    case GUI_MSG_PLAY_TRAILER:
+    {
+      const CFileItemPtr item = std::static_pointer_cast<CFileItem>(message.GetItem());
+      CFileItem* trailerItem =
+          ContentUtils::GeneratePlayableTrailerItem(item, g_localizeStrings.Get(20410)).release();
```

Comment:
Close, but there's still a mem leak. Ideally, you're using `release()` correctly if the raw pointer is exposed for as little as possible (when it's returned). So you would want to assign to `std::unique_ptr<CFileItem> trailerItem =`. That way, when it's not released below, it will be garbage collected as it falls out of scope.

Reply:
thanks for the class :)
hopefully I did it right this time